engine/sound/src/decoders/decoder_opus.cpp

```diff
@@ -0,0 +1,502 @@
+// Copyright 2020-2024 The Defold Foundation
+// Copyright 2014-2020 King
+// Copyright 2009-2014 Ragnar Svensson, Christian Murray
+// Licensed under the Defold License version 1.0 (the "License"); you may not use
+// this file except in compliance with the License.
+//
+// You may obtain a copy of the License, together with FAQs at
+// https://www.defold.com/license
+//
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+#include <stdint.h>
+#include <dlib/array.h>
+#include <dlib/index_pool.h>
+#include <dlib/log.h>
+#include <dlib/math.h>
+#include <dlib/profile.h>
+
+#include "opus/opus.h"
+#include "sound_codec.h"
+#include "sound_decoder.h"
+
+#include <string.h>
+
+
+#define STREAM_BLOCK_SIZE (16 << 10)
+
+namespace dmSoundCodec
+{
+    namespace
+    {
+        struct Decodestream_info {
+            Info m_Info;
+            uint32_t m_stream_serial;
+            uint32_t m_SamplePos;
+            uint32_t m_DecodeSamplePos;
+            uint16_t m_PreSkip;
+            float m_SampleScale;
+            OpusDecoder* m_Decoder;
+            dmSound::HSoundData m_SoundData;
+            uint32_t m_StreamOffset;
+            dmArray<uint8_t> m_DataBuffer;
+            uint32_t m_LastOutputOffset;
+            dmArray<uint8_t> m_LastOutput;
+            uint8_t m_lacing_table[255];
+            uint8_t m_lacing_tableSize;
+            uint32_t m_lacing_tableIndex;
+            uint32_t m_SkipBytes;
+            uint8_t m_PacketBuffer[2048];
+            uint32_t m_PacketOffset;
+            bool m_bEOS : 1;
+        };
+    }
+
+    static void CleanupBuffer(dmArray<uint8_t>& buffer, uint32_t consumed)
+    {
+        // Move sill un-consumed buffer contents to the front so we offer them in one contigous chunk to Vorbis at all times
+        if (consumed == 0)
+            return;
+
+        assert(consumed <= buffer.Size());
+
+        uint32_t n = buffer.Size();
+        uint8_t* data = buffer.Begin();
+        for(uint32_t i=consumed; i<n; ++i)
```

Comment:
[Codestyle] Spaces between variables and operators; between for and open bracket.
If condition's body is single statement - it should be wrapped with {}